### ISSUE TITLE
urlPattern : use regex group

### ISSUE THREAD

**Issue by ptitou** [tomakehurst/wiremock#1038]
## Feature requests

In my stub, I'd like to reuse a part of url (matching a regex) to use it in response statement.

my stub : 
{
    "name": "get",
    "request": {
        "urlPattern": "/rest/api/2/issue/(\w+)-\d+.*",
        "method": "GET"
    },
    "response": {
        "status": 200,
        "bodyFileName": "issue_{{request.requestLine.matches[0]}}.json"
    }
} 

/rest/api/issue/JIRA-14 return body file issue_JIRA.json
/rest/api/issue/TESTS-1256 return body file issue_TESTS.json

**Comment by abinayasakthi** [comment]
Hi @tomakehurst I was trying to fix this issue. Could you throw some idea on the fix ?

**Comment by tomakehurst** [comment]
Do you mean you're planning to create a pull request?

**Comment by abinayasakthi** [comment]
Yes

**Comment by tomakehurst** [comment]
Unfortunately, with the way things are factored at the moment this isn't possible.

Because the URL regex lives in the `RequestPattern` (under `StubMapping`) object, it's not visible to transformers (and the response templating component is implemented as a transformer). Since transformers are part of the public API, changing this would need to be done pretty carefully.

**Comment by abinayasakthi** [comment]
Yes, my understanding was same as you quoted. Thanks @tomakehurst. Will look for other issues.

**Comment by tomakehurst** [comment]
It's a shame, as it would be a useful thing to have.

Thanks for the offer to contribute, anyway!

**Comment by verhage** [comment]
Sorry for commenting on this old issue, but it might be of use to someone.

Although it would be a nice feature you don't actually need capturing groups in the urlPattern to achieve what you want. There's another approach accessing the request from within the response template.

For example, I had this case where I needed to mock an API and the URL contained a customer_id as c_1234. I needed only the numeric part of this id, 1234 in the response.

In my response template, I managed this using a regexExtract on the request URL:

```
{{ regexExtract request.pathSegments.[1] '\\d+' }}
```
